### DISCUSSION TITLE
Question about the output

### DISCUSSION THREAD

**Discussion by smallrainbow** [SmileiPIC/Smilei#769] [General]
Dear developers,

Coud you so kind to answer the following questions on output, please?

(1) Does smilei have a keyword in the namelist to control the output precision? For example, double precision, single precision (in order to save storage space as much as possible when saving particle data).

(2) In th block "DiagTrackParticles", what keyword can be used to save the particle ID?

(3) In the block "DiagParticleBinning", the pressure tensor expression output by the keyword weight_vx_px is Pxx=mass*weight*px^2/sqrt(1+px^2+py^2+pz^2) in /src/Diagnostic/Histogram.h.
Is there a way to output the pressure tensor expression on page 6 as shown below?
https://ocw.mit.edu/courses/22-611j-introduction-to-plasma-physics-i-fall-2003/05a1bba54c1ba9ff2208355d70120ce0_chap4.pdf

Thank you very much!

Best regards,
Zhongwei

**Comment by smallrainbow** [comment]
I see, both of ID and precision can be defined in the namelist.
And the pxx is similar to that in vpic. I have no nessesary to save pressure tensor because the calculation does not use it (if we do it, the code performence will decrease). The output of pxx,pyy,pxy etc. already enought for calculating any pressure tensor component. 
Sorry for the questions above. Everything is fine now. Thanks.

**Comment by mccoys** [comment]
1) precision is only available for scalar diagnostics
2) I believe that the id is always saved anyways so that the order can be reconstructed
3) for different expressions of the pressure tensor, please note that custom values can be done. Check also this: https://github.com/SmileiPIC/Smilei/discussions/773#discussioncomment-11374833
